src/engine/effects/engineeffect.cpp

```diff
@@ -17,9 +20,25 @@ EngineEffect::EngineEffect(const EffectManifest& manifest,
         m_parametersById[parameter.id()] = pParameter;
     }
 
+    for (const ChannelHandleAndGroup& inputChannel :
+            pEffectsManager->registeredInputChannels()) {
+        ChannelHandleMap<EffectEnableState> outputChannelMap;
+        for (const ChannelHandleAndGroup& outputChannel :
+                pEffectsManager->registeredOutputChannels()) {
+            outputChannelMap.insert(outputChannel.handle(), EffectEnableState::Disabled);
+        }
+        m_effectEnableStateForChannelMatrix.insert(inputChannel.handle(), outputChannelMap);
```

Comment:
We enable an effect only by input channel. So we need no matrix here.

Reply:
No, every combination of input and output channel needs its own independent state. See the `process` method.

Reply:
Yes, I understand. But you can always create the output channels for one input channel together. The inner hash container.

Reply:
I don't understand what you mean. Could you elaborate on your idea of how this could be changed to not use a matrix?

Reply:
We have the outer array with index of input channel and the inner array with index of output channel. This inner array can be passed together with the two channels.

Reply:
I disagree. The code that calls this constructor should not have anything to do with EffectEnableStates.